docs/rules/display-name.md

```diff
@@ -1,5 +1,7 @@
 # Disallow missing displayName in a React component definition (react/display-name)
 
+✔️ This rule is part of the [recommended](https://github.com/jsx-eslint/eslint-plugin-react#recommended) config.
```

Comment:
i'm actually not too keen on advertising this; plugins shipping configs is imo a mistake, because the semver of the config forces a breaking change on an otherwise semver-minor plugin/option addition. I don't want to imply that rules not in the recommended config aren't recommended - a great many of them are.

Reply:
I strongly recommend (pun not intended haha) that we include it and I'll share my reasoning below:

1. The `recommended` config exists today. Many users use it and need to know what's in it (i.e. which rules are applying to their codebase because of it). These notices (which are used by ESLint and countless plugins) serve that purpose of improving clarity and usability.
2. If the config is removed someday because of your semver concerns, the notices would be removed too of course. 
3. Realistically, users prefer the convenience of enabling a "generally safe / acceptable" config rather than having to hand-pick from a hundred rules themselves.
4. Personally, I haven't heard this concern of a non-`recommended`-config rule being thought of as "not recommended" or "discouraged". That seems like a bit of a stretch to me and if there was confusion we could explain in the documentation why some rules are included in `recommended` and others aren't.

That said, if you won't accept it, I'll remove it or we can close the PR.

Reply:
To be clear, if we ever do a semver major, I’d remove the recommended config entirely - from any plugin i maintain - because I’ve found it to be so problematic in this way.

Reply:
Okay, while I personally wouldn't agree with that, I'll close this PR due to lack of interest.